### PR TITLE
Update unity requirement to 2018.3

### LINKED ISSUE #3192: UWP apps built via command line (without opening Unity Editor) after a clean do not launch
**Does this affect the legacy HoloToolkit (master) or the Mixed Reality Toolkit (mrtk_release)?**
Mixed Reality Toolkit

**Describe the bug**
When building a project which uses MRTK from the command line after a clean of any Unity Library files, a built UWP does not launch.  We seem to be getting out of bounds serialization errors from levels/shared assets.  An empty scene seemed to be able to launch, but once I begin adding MRTK dependencies, I get issues.

I have tracked the issue down, and it appears to be related to the Microsoft.MixedReality.Toolkit.Core.Utilities.Editor.Setup.dll.  When I delete this dll and remove the . from in front of MixedRealityEditorSettings.cs, I am able to clean, and build from the command line and successfully launch the app.

### LINKED ISSUE #3306: Update Unity requirement to 2018.3
## Overview
To get the Mixed Reality Toolkit closer to the upcoming Unity 2018 LTS release, it will need to update to 2018.3 and drop support for 2018.1 and 2018.2.

This will be a breaking change as Unity has changed prefab formats in 2018.3.

## Requirements
- [x] Check for 2018.3 or newer on
    - [x] Project load
    - [x] Package import
- [x] Display a message indicating an unsupported version is being used and allow user to:
    - [x] Continue (and expect errors / incorrect behavior)
    - [x] Close the editor
- [x] Project loads without script warnings
- [X] Apps behave as previously

### PR DESCRIPTION
**Pull request by davidkline-ms**
Overview
---
This change brings the required Unity version to 2018.3.x.

PLEASE NOTE: It is recommended that you delete your Library folder after pulling this change.

Changes
---
- Fixes #3306
- Fixes #3192

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿using UnityEditor;
+
+#if UNITY_EDITOR
+
+[InitializeOnLoad]
+internal class EditorProjectUtilities
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs]: Missing namespace / license header / etc

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs]: good catch, thanks!

File: Assets/MixedRealityToolkit/_Core/Utilities/BuildAndDeploy/UwpBuildDeployWindow.cs [hunk 2]
```diff
@@ -288,7 +288,8 @@ private void OnGUI()
 
             if (GUILayout.Button("Open Player Settings"))
             {
-                EditorApplication.ExecuteMenuItem("Edit/Project Settings/Player");
+                // TODO: Remove use of "Unsupported" class.
```
Comment (chrisfromwork) [inline Assets/MixedRealityToolkit/_Core/Utilities/BuildAndDeploy/UwpBuildDeployWindow.cs]: do we want issues tracking todos?

File: Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        /// <summary>
+        /// Checks that a supported version of Unity is being used with this project.
+        /// </summary>
+        /// <remarks>
+        /// This method displays a message to the user allowing them to continue or to exit the editor.
+        /// </remarks>
+        public static void CheckMinimumEditorVersion()
+        {
+#if !UNITY_2018_3_OR_NEWER
+        DisplayIncorrectEditorVersionDialog();
```
Comment (chrisfromwork) [inline Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs]: nit: missing tab/spaces

File: Assets/MixedRealityToolkit/_Core/Utilities/BuildAndDeploy/UwpBuildDeployWindow.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/_Core/Utilities/BuildAndDeploy/UwpBuildDeployWindow.cs]: yes please :)

File: Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs (hunk 3 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs]: thanks!

File: Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEditor;
+
+#if UNITY_EDITOR
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs]: This can be removed since it's in an editor assembly

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/_Core/Utilities/Editor/EditorProjectUtilities.cs]: done  :)

**Pull request merged**